iceoryx_hoofs/design/include/iox/move_and_copy_helper.hpp

```diff
@@ -37,85 +35,45 @@ enum class MoveAndCopyOperations
 /// operation type (Opt).
 /// @tparam Opt The operation type that determines how objects are created or assigned.
 template <MoveAndCopyOperations Opt>
-struct MoveAndCopyHelper
+class MoveAndCopyHelper final
 {
+  public:
     /// @brief Creates or assigns an object to 'dest' based on the specail operation type.
     /// @tparam T The type of the object to be created or assigned.
     /// @tparam V The type of the source object, kept as a universal reference to preserve its lvalue or rvalue nature.
     /// @param[out] dest The destination object where the new object is created or to which the source object is
     /// assigned.
     /// @param[in] src The source object, either for copy or move operations.
     template <typename T, typename V>
-    static inline void transfer(T& dest, V&& src) noexcept
-    {
-        if constexpr (is_ctor())
-        {
-            ctor_create(dest, std::forward<V>(src));
-        }
-        else
-        {
-            assignment_create(dest, std::forward<V>(src));
-        }
-    }
+    static void transfer(T& dest, V&& src) noexcept;
 
     /// @brief Force to use constructor to create an object at the destination.
     /// @tparam T The type of the object to be constructed.
     /// @tparam V The type of the source object, used for move or copy construction.
     /// @param[out] dest The destination object where the new object is constructed.
     /// @param[in] src The source object, either for move or copy construction.
     template <typename T, typename V>
-    static inline void ctor_create(T& dest, V&& src) noexcept
-    {
-        if constexpr (is_move())
-        {
-            static_assert(std::is_rvalue_reference_v<decltype(src)>, "src should be rvalue reference");
-            static_assert(std::is_convertible_v<V, T>, "src type is not convertible to dest type");
-            new (&dest) T(std::forward<V>(src));
-        }
-        else
-        {
-            static_assert(std::is_lvalue_reference_v<decltype(src)>, "src should be lvalue reference");
-            static_assert(std::is_const_v<std::remove_reference_t<decltype(src)>>, "src should has 'const' modifier");
-            static_assert(std::is_convertible_v<V, T>, "src type is not convertible to dest type");
-            new (&dest) T(src);
-        }
-    }
+    static void ctor_create(T& dest, V&& src) noexcept;
```

Comment:
For example, when we use copy assignment for `FixedPositionContainer`, the `FREE` slot may remain uninitialized. In this case, we are forecd to use `ctor_create`.

Reply:
Right, totally forgot about that. Can you document when the three methods should be used?

As far as I understand it
- `transfer` is what you usually want/need if you don't work with uninitialized memory
- `ctor_create` is what you might need for low level building blocks when working with uninitialized memory and the copy/move ctor must be used for members instead of the assignment operator, even when used from the assignment operator or the type
- `assignment_create` ... for this one I don't find a scenario

What do you think of renaming `ctor_create` to `create_new` and `assignment_create` to `assign`?

Reply:
Of course, I'll rename these two functions.